docs/tileentities/tileentity.md

```diff
@@ -0,0 +1,128 @@
+    
+## Synchronizing the Data to the Client
+There are 3 (three) ways of syncing data to the client.
+Synchronizing on chunk load, synchronizing on block updates and synchronizing with a custom network message.
+
+### Synchronizing on chunk load
+For this you need to override
+```JAVA
+    TileEntity#getUpdateTag()
+
+    TileEntity#handleUpdateTag(NBTTagCompound nbt)
+```
+Again, this is pretty simple, the first method collects the data that should be send to the client.
+While the second one processes that data. If your `TileEntity` doesn't contain much data you might be able to use the methods out of the `Storing Data within your TileEntity` section.
+!!! note
+    Synchronizing excessive/useless data for TileEntities can lead to network congestion. You should optimize your network usage by sending only the information the client needs when the client needs it. For instance, it is more often that not unnecessary to send the inventory of a tile entity in the update tag, as this can be synchronized via its GUI.
+
+### Synchronizing on block update
+This method is a bit more complicated, but again you just need to override 2 methods.
+Here is a tiny example implementation of it 
+```JAVA
+    @Override
+    public SPacketUpdateTileEntity getUpdatePacket(){
+        NBTTagCompound nbtTag = new NBTTagCompound();
+        //Write your data into the nbtTag
+        return new SPacketUpdateTileEntity(getPos(), 1, nbtTag);
+    }
+
+    @Override
+    public void onDataPacket(NetworkManager net, SPacketUpdateTileEntity pkt){
+        NBTTagCompound tag = pkt.getNbtCompound();
+        //Handle your Data
+    }
+```
+The Constructor of `SPacketUpdateTileEntity` takes:
+The position of your `TileEntity`.
+An ID, though it isn't really used besides by Vanilla, therefore you can just put a 1 in there.
+An `NBTTagCompound` which should contain your data.
+Additionally to this you now need to cause a "BlockUpdate" on the Client.
+```JAVA
+    World#notifyBlockUpdate(BlockPos pos, IBlockState oldState, IBlockState newState, int flags)
+```
+The `pos` should be your TileEntitiy's position. For `oldState` and `newState` you can pass the current BlockState at that position. The `flags` should be 2, which will sync the changes to the clien
+
+### Synchronizing using a custom network message
+This way of synchronizing is probably the most complicated one.
+You should first check out the `Networking` section and specially [`SimpleImpl`][SimpleImpl] before attempting this.
+Once you've created your custom network message, you can send it to all users that have the `TileEntity` loaded with:
+```JAVA
+    SimpleNetworkWrapper#sendToAllTracking(IMessage, NetworkRegistry.TargetPoint)
+```
+!!! note
+    It is important that you do safety checks, the TileEntity might already be destroyed/replaced when the message arrives at the player!
+    You should also check if the chunk is loaded (`World#isBlockLoaded`()`)
```

Comment:
You should also check if the chunk is loaded with `World#isBlockLoaded(BlockPos)`.